src/image/Image.cc

```diff
@@ -697,6 +698,20 @@ geom::Box2I bboxFromMetadata(daf::base::PropertySet & metadata)
     return geom::Box2I(xy0, dims);
 }
 
+template <typename T1, typename T2>
+bool imagesOverlap(ImageBase<T1> const& image1, ImageBase<T2> const& image2) {
+
+    auto arr1 = image1.getArray();
+    auto beg1Addr = reinterpret_cast<void const *>(arr1.front().begin());
+    auto end1Addr = reinterpret_cast<void const *>(arr1.back().end());
```

Comment:
I thought I needed `reinterpret_cast` but you are right that I can do without it (the compiler was complaining about something else).

I agree that `arr1.front().begin()` is less than obvious, but it's what @TallJimbo suggested and I think it's safer than `arr[height][width]` because one can easily get the indices in the wrong order. I will document what it does, but I can't think of anything else to do about it.

Reply:
I think if you use an implicit conversion to `void*` than there's no need to document that construct -- pointer to element is the only thing that both compiles and makes sense.